updater/updater_test.go

```diff
@@ -0,0 +1,179 @@
+}
+
+func newForTest(t *testing.T) *api.API {
+	if _, ok := os.LookupEnv("NEBRASKA_DB_URL"); !ok {
+		log.Printf("NEBRASKA_DB_URL not set, setting to default %q\n", defaultTestDbURL)
+		_ = os.Setenv("NEBRASKA_DB_URL", defaultTestDbURL)
+	}
+	a, err := api.New(api.OptionInitDB)
+
+	require.NoError(t, err)
+	require.NotNil(t, a)
+
+	return a
+}
+
+func newUpdater(a *api.API, appID string, channel string, instanceID string, instanceVersion string) *Updater {
+	return &Updater{
+		omahaURL:        "http://localhost:8000",
+		clientVersion:   "go-omaha",
+		instanceID:      instanceID,
+		sessionID:       uuid.New().String(),
+		appID:           appID,
+		instanceVersion: instanceVersion,
+		channel:         channel,
+		omahaReqHandler: NewTestHandler(a),
+	}
+}
+
+func TestNewUpdater(t *testing.T) {
+	_, err := New("http://localhost:8000", "io.phony.App", "stable", "instance001", "0.1.0")
+	assert.NoError(t, err)
+}
+
+func TestCheckForUpdates(t *testing.T) {
+	a := newForTest(t)
+	defer a.Close()
+
+	tTeam, _ := a.AddTeam(&api.Team{Name: "test_team"})
+	tApp, _ := a.AddApp(&api.Application{Name: "io.phony.App", TeamID: tTeam.ID})
+	tPkg, _ := a.AddPackage(&api.Package{Type: api.PkgTypeOther, URL: "http://sample.url/pkg", Version: "0.1.0", ApplicationID: tApp.ID, Arch: api.ArchAMD64})
+	tChannel, _ := a.AddChannel(&api.Channel{Name: "channel1", Color: "blue", ApplicationID: tApp.ID, PackageID: null.StringFrom(tPkg.ID), Arch: api.ArchAMD64})
+	tGroup, err := a.AddGroup(&api.Group{Name: "group1", ApplicationID: tApp.ID, ChannelID: null.StringFrom(tChannel.ID), PolicyUpdatesEnabled: true, PolicySafeMode: true, PolicyPeriodInterval: "15 minutes", PolicyMaxUpdatesPerPeriod: 2, PolicyUpdateTimeout: "60 minutes", Track: "stable"})
+	assert.NoError(t, err)
+
+	u := newUpdater(a, tApp.ID, tGroup.Track, "instance-0001", "0.2.0")
+	info, err := u.CheckForUpdates(context.TODO())
+	assert.NoError(t, err)
+	assert.False(t, info.HasUpdate)
+	assert.Equal(t, "", info.GetVersion())
+
+	newPkg, _ := a.AddPackage(&api.Package{Type: api.PkgTypeOther, URL: "http://sample.url/pkg", Version: "0.3.0", ApplicationID: tApp.ID, Arch: api.ArchAMD64, Filename: null.StringFrom("updatefile.txt")})
+	tChannel.PackageID = null.StringFrom(newPkg.ID)
+	err = a.UpdateChannel(tChannel)
+	assert.NoError(t, err)
+
+	info, err = u.CheckForUpdates(context.TODO())
+	assert.NoError(t, err)
+	assert.True(t, info.HasUpdate)
+
+	version := info.GetVersion()
+	assert.Equal(t, "0.3.0", version)
+
+	urls := info.GetURLs()
+	assert.NotNil(t, urls)
+	assert.Equal(t, 1, len(urls))
+	assert.Equal(t, urls[len(urls)-1], info.GetURL())
+	assert.Equal(t, "http://sample.url/pkg", info.GetURL())
+
+	pkg := info.GetPackage()
+	assert.NotNil(t, pkg)
+	assert.Equal(t, "updatefile.txt", pkg.Name)
+}
+
+type updateTestHandler struct {
+	fetchUpdateResult error
+	applyUpdateResult error
+}
+
+func (u updateTestHandler) FetchUpdate(ctx context.Context, info *UpdateInfo) error {
+	// Nothing to do
```

Comment:
I saw other comments ending with `.` and it's a good practice, so it would be nice to be consistent.
```suggestion
	// Nothing to do.
```

Or even better, remove the comment :wink: IMO it does not add any valuable context.